meta-ostro/lib/supportedrecipes/supportedrecipes.py

```diff
@@ -0,0 +1,346 @@
+                    return True
+            return False
+
+        def find_collections(self, pn):
+            collections = set()
+            for supported_recipe in self.supported:
+                if supported_recipe.supported(pn, None):
+                    collections.add(supported_recipe.collection_re[1])
+            return collections
+
+    files = []
+    supported_files = d.getVar('SUPPORTED_RECIPES', True)
+    if not supported_files:
+        bb.fatal('SUPPORTED_RECIPES is not set')
+    supported_recipes = SupportedRecipes()
+    for filename in supported_files.split():
+        try:
+            with open(filename) as f:
+                linenumber = 1
+                for line in f.readlines():
+                    if line.startswith('#'):
+                        continue
+                    # TODO (?): sanity check the content to catch obsolete entries or typos.
+                    pn = line.strip()
+                    if pn:
+                        supported_recipes.append(SupportedRecipe(line.strip(), filename, linenumber))
+                    linenumber += 1
+            files.append(filename)
+        except OSError as ex:
+            bb.fatal('Could not read SUPPORTED_RECIPES = %s: %s' % (supported_files, str(ex)))
+
+    return (supported_recipes, files)
+
+SOURCE_FIELDS = 'component,collection,version,homepage,source,summary'.split(',')
+
+# Collects information about one recipe during parsing for SUPPORTED_RECIPES_SOURCES.
+# The dumped information cannot be removed because it might be needed in future
+# bitbake invocations, so the default location is inside the tmp directory.
+def dump_sources(d):
+    pn = d.getVar('PN', True)
+    filename = d.getVar('FILE', True)
+    collection = bb.utils.get_file_layer(filename, d)
+    pv = d.getVar('PV', True)
+    summary = d.getVar('SUMMARY', True) or ''
+    homepage = d.getVar('HOMEPAGE', True) or ''
+    src = d.getVar('SRC_URI', True).split()
+    sources = []
+    for url in src:
+        scheme, netloc, path, query, fragment = urlparse.urlsplit(url)
+        if scheme != 'file':
+            parts = path.split(';')
+            if len(parts) > 1:
+                path = parts[0]
+                params = dict([x.split('=') if '=' in x else (x, '') for x in parts[1:]])
+            else:
+                params = {}
+            name = params.get('name', None)
+            sources.append((name, '%s://%s%s' % (scheme, netloc, path)))
+    dumpfile = d.getVar('SUPPORTED_RECIPES_SOURCES_DIR', True) + '/' + pn + filename
+    bb.utils.mkdirhier(os.path.dirname(dumpfile))
+    with open(dumpfile, 'w') as f:
+        # File intentionally kept small by not writing a header line. Guaranteed to contain SOURCE_FIELDS.
+        writer = csv.writer(f)
+        for idx, val in enumerate(sources):
+            name, url = val
+            if name and not len(sources) == 1:
+                fullname = '%s/%s' % (pn, name)
```

Comment:
Not in this case, pn and name are not filename components. This implements the <recipe>[/<source name>] special case for recipes with multiple components.